src/vivarium_gates_mncnh/components/intervention.py

```diff
@@ -308,12 +309,26 @@ def update_hemoglobin_exposure(
         self, index: pd.Index, exposure: pd.Series[float]
     ) -> pd.Series[float]:
         pop = self.population_view.get(index)
-        exposure.loc[
-            pop[COLUMNS.ORAL_IRON_INTERVENTION] == models.ORAL_IRON_INTERVENTION.NO_TREATMENT
-        ] -= (self.ifa_coverage * self.ifa_effect_size)
-        exposure.loc[
+
+        has_first_trimester_anc = pop[COLUMNS.ANC_ATTENDANCE].isin(
+            [
+                ANC_ATTENDANCE_TYPES.FIRST_TRIMESTER_ONLY,
+                ANC_ATTENDANCE_TYPES.FIRST_TRIMESTER_AND_LATER_PREGNANCY,
+            ]
+        )
+        has_later_pregnancy_anc = (
+            pop[COLUMNS.ANC_ATTENDANCE] == ANC_ATTENDANCE_TYPES.LATER_PREGNANCY_ONLY
```

Comment:
Why do you distinguish between first trimester and later? You do the same thing for both.

Reply:
From Ali: "The improvement is that it handles the first trimester effects separately from the later pregnancy ANC visit effects. This is important because the oral iron-modified exposure will be used to determine IV iron eligibility at the later pregnancy visit."